### PR TITLE
Support libsonnet file extensions

### PR DESCRIPTION
**Pull request by bryant1410**
Add "libsonnet" extension, which is the default template extension. I saw the way to do it in https://github.com/JetBrains/intellij-community/blob/master/plugins/yaml/src/org/jetbrains/yaml/YAMLFileTypeLoader.java

### CODE REVIEW

File: src/com/jsonnetplugin/JsonnetFileTypeFactory.java [hunk 1]
```diff
@@ -6,34 +6,6 @@
...
-                    @Override
-                    public String getPresentableString() {
-                        return ".jsonnet";
-                    }
-                },
-                new FileNameMatcher(){
-
-                    @Override
-                    public boolean accept(@NotNull String fileName) {
-                        return fileName.endsWith(".jsonnet.TEMPLATE");
```
Comment (lihaoyi-databricks) [inline src/com/jsonnetplugin/JsonnetFileTypeFactory.java]: Sorry I'm just coming to look at this; wouldn't this change mean `.jsonnet.TEMPLATE` extension is no longer supported? That's the naming convention we use internally

Comment (bryant1410) [reply src/com/jsonnetplugin/JsonnetFileTypeFactory.java]: Yes, it's not supported anymore. Why don't you use [the one recommended by Jsonnet, "libsonnet"](https://jsonnet.org/learning/tutorial.html#imports)?

> Files designed for import by convention end with .libsonnet

I haven't seen "jsonnet.TEMPLATE" anywhere else.

**Comment by bryant1410** [comment]
#22 shows another example of "libsonnet" common usage. Can you guys consider merging this PR?

**Review by JoshRosen — COMMENTED** [review]
@bryant1410, it looks like the current version of this PR removes support for the `.jsonnet.TEMPLATE` extension. That removal will break Databricks' internal use of this plugin: migrating to `.libsonnet` would be time-consuming and non-trivial for us (we'd have to rename thousands of files in multiple git branches, plus make changes to other internal tools).

Could you revise this PR so that it adds support for `.libsonnet` while preserving support for the existing `.jsonnet.TEMPLATE` file extension? If you do that, could you also update the first sentence of `README.md` to read

> This plugin teaches IntelliJ how to understand `.jsonnet`, `.jsonnet.TEMPLATE`, and `.libsonnet` files. It includes the following features:

I'm +1 towards improving this plugin's out-of-the-box user experience, but also want to ensure that we don't regress that experience for Databricks' internal users.

**Comment by lihaoyi-databricks** [comment]
I just tried PR branch out locally; it doesn't seem to be recognizing my `.jsonnet.TEMPLATE` files. Maybe we need to use the verbose `new FileNameMatcher()` to preserve that functionality?

**Comment by bryant1410** [comment]
I didn't try the latest change locally, and don't remember how I tested the previous one. How can I try locally?

**Comment by lihaoyi-databricks** [comment]
I left a comment here https://github.com/databricks/intellij-jsonnet/issues/19#issuecomment-576872332 with some pointers, IIRC it's following the "Plugin Devkit" workflow here https://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/using_dev_kit.html

(I do not like this workflow, but it is what it is. There's a new gradle-based workflow that apparently works, would welcome a PR if anyone wants to help set that up! https://www.jetbrains.org/intellij/sdk/docs/tutorials/build_system.html)

**Comment by bryant1410** [comment]
@lihaoyi-databricks it should be good to go now

**Comment by lihaoyi-databricks** [comment]
Thanks!

**Comment by ddragoti** [comment]
Can we get a new build of this plugin which includes the latest merged changes? Seems like the one in the Marketplace (1.1) is from 8 months ago.

**Comment by GillesJ** [comment]
+1 for updating the build of this plugin in the Marketplace.

Currently .libsonnet does not work.

**Comment by pmorch** [comment]
This was merged more than a year ago and no release has been made since, which is a shame. Please release this! :-)

Especially since building this locally [apparently isn't straightforward](https://github.com/databricks/intellij-jsonnet/issues/19#issuecomment-576872332).

**Comment by farquet** [comment]
For any reader awaiting a release to get `.libsonnet` files recognized, note that it's straightforward to manually add a file pattern be recognized as jsonnet until this happens.
In IntelliJ, do:
`
Preferences -> File Types -> Scroll to 'Jsonnet language file' -> Add '*.libsonnet' to the name patterns
`

**Pull request merged**